flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -1828,6 +1828,33 @@ class JobManager(
     jobManagerMetricGroup.gauge[Long, Gauge[Long]]("numRunningJobs", new Gauge[Long] {
       override def getValue: Long = JobManager.this.currentJobs.size
     })
+    jobManagerMetricGroup.gauge[Long, Gauge[Long]]("numFailedJobs", new Gauge[Long] {
+      override def getValue: Long = {
+         var failedJobs = 0
+         val ourJobs = createJobStatusOverview()
+         val future = (archive ? RequestJobsOverview.getInstance())(timeout)
+         val archivedJobs : JobsOverview = Await.result(future, timeout).asInstanceOf[JobsOverview]
+         failedJobs += ourJobs.getNumJobsFailed() + archivedJobs.getNumJobsFailed()
+         failedJobs
+    }})
+    jobManagerMetricGroup.gauge[Long, Gauge[Long]]("numCancelledJobs", new Gauge[Long] {
+      override def getValue: Long = {
+         var cancelledJobs = 0
+         val ourJobs = createJobStatusOverview()
+         val future = (archive ? RequestJobsOverview.getInstance())(timeout)
+         val archivedJobs : JobsOverview = Await.result(future, timeout).asInstanceOf[JobsOverview]
+         cancelledJobs += ourJobs.getNumJobsCancelled() + archivedJobs.getNumJobsCancelled()
+         cancelledJobs
+    }})
+    jobManagerMetricGroup.gauge[Long, Gauge[Long]]("numFinishedJobs", new Gauge[Long] {
+      override def getValue: Long = {
+         var finishedJobs = 0
+         val ourJobs = createJobStatusOverview()
+         val future = (archive ? RequestJobsOverview.getInstance())(timeout)
+         val archivedJobs : JobsOverview = Await.result(future, timeout).asInstanceOf[JobsOverview]
+         finishedJobs += ourJobs.getNumJobsFinished() + archivedJobs.getNumJobsFinished()
+         finishedJobs
+    }})
```

Comment:
i think that would result in too much additional bookkeeping for the JobManager; as i understand it we would effectively copy a lot of the Archivist functionality into the JobManager.

One simple way to reduce the # of RPC calls from 3 to 1 is something like this:
```
public class RpcSavingContainer {
	private JobsOverview overview;
	private int count = 0;
	
	public synchronized JobsOverview getOverview() {
		if (count == 0) {
			overview = // result of RPC call
		}
		count = count++ % 3;
		return overview;
	}
}
```
Each gauge would use the same instance of this container.

Another option would be to instantiate these metrics in the MemoryArchivist and not the JobManager.